fedimint-dummy-server/src/lib.rs

```
@@ -0,0 +1,463 @@
+            .cfg
+            .consensus
+            .public_key_set
+            .public_key_share(peer_id.to_usize())
+            .verify(&share.0, request.clone())
+        {
+            bail!("Signature share is invalid");
+        }
+
+        dbtx.insert_new_entry(&DummySignatureShareKey(request.clone(), peer_id), &share)
+            .await;
+
+        // Collect all valid signature shares previously received
+        let signature_shares = dbtx
+            .find_by_prefix(&DummySignatureShareStringPrefix(request.clone()))
+            .await
+            .collect::<Vec<_>>()
+            .await;
+
+        if signature_shares.len() <= self.cfg.consensus.public_key_set.threshold() {
+            return Ok(());
+        }
+
+        let threshold_signature = self
+            .cfg
+            .consensus
+            .public_key_set
+            .combine_signatures(
+                signature_shares
+                    .iter()
+                    .map(|(peer_id, share)| (peer_id.1.to_usize(), &share.0)),
+            )
+            .expect("We have verified all signature shares before");
+
+        dbtx.remove_by_prefix(&DummySignatureShareStringPrefix(request.clone()))
+            .await;
+
+        dbtx.insert_entry(
+            &DummySignatureKey(request.to_string()),
+            &Some(SerdeSignature(threshold_signature)),
+        )
+        .await;
+
+        Ok(())
+    }
+
+    fn build_verification_cache<'a>(
+        &'a self,
+        _inputs: impl Iterator<Item = &'a DummyInput> + Send,
+    ) -> Self::VerificationCache {
+        DummyVerificationCache
+    }
+
+    async fn process_input<'a, 'b, 'c>(
+        &'a self,
+        dbtx: &mut ModuleDatabaseTransaction<'c>,
+        input: &'b DummyInput,
+        _cache: &Self::VerificationCache,
+    ) -> Result<InputMeta, ModuleError> {
+        let current_funds = dbtx
+            .get_value(&DummyFundsKeyV1(input.account))
+            .await
+            .unwrap_or(Amount::ZERO);
```

Is there a case here where unwrapping would be unwise since it masks an underlying error?